spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/config/LocalResponseCacheAutoConfiguration.java

```diff
@@ -45,8 +44,7 @@
 @Configuration(proxyBeanMethods = false)
 @EnableConfigurationProperties({ LocalResponseCacheProperties.class })
 @ConditionalOnClass({ Weigher.class, Caffeine.class, CaffeineCacheManager.class })
-@ConditionalOnProperty(name = "spring.cloud.gateway.enabled", matchIfMissing = true)
-@ConditionalOnEnabledFilter(LocalResponseCacheGatewayFilterFactory.class)
```

Comment:
Why the change from the enabled filter to annotation to a config prop

Reply:
It's an important error, if we base it on the `LocalResponseCacheGatewayFilterFactory` the cache doesn't work for every route using the [global property as we indicate on the docs](https://docs.spring.io/spring-cloud-gateway/docs/current/reference/html/#local-cache-response-global-filter). Right now is just working for those endpoints configured at route level 🤦🏼‍♀️  and we must base it on the property to instantiate it globally.

Reply:
Ok. The conditional on gateway enabled should probably stay

Reply:
Totally agree, I got carried away with this fix of the cache functionality: https://github.com/spring-cloud/spring-cloud-gateway/pull/2848/commits/61fe95109f7089af71b5cad8e7087e05d6993cf9

Thanks for your patient!

Reply:
Totally agree, I got carried away with this fix of the cache functionality: https://github.com/spring-cloud/spring-cloud-gateway/pull/2848/commits/0dd1d831f232645dad5ac61df53471221f3e71d2

Thanks for your patient!